src/components/singleTopicPage.jsx

```diff
@@ -4,28 +4,38 @@ import {useParams} from 'react-router-dom'
 import {Link} from 'react-router-dom'
 import { getAllArticles } from '../utils/api';
 
-function SingleTopicPage() {
+function SingleTopicPage({sortBy, setSortBy, orderBy, setOrderBy}) {
     const [topicArticles, setTopicArticles] = useState([])
 
+
     const {topic} = useParams();
 
     useEffect(() => {
-        getAllArticles()
+        getAllArticles(topic, sortBy, orderBy)
         .then(({data : {articles}})=> {
+
             const newArticles = articles.filter((article) => {
                 if (article.topic === topic){
-                    console.log('here')
+
                     return {...article}
-                }
+                } 
             })
             setTopicArticles(newArticles)
         })
-    }, [])
+    }, [sortBy, orderBy] )
```

Comment:
This dependancy array is missing "topic" it should show an error in your terminal when you 'npm run start'